### ISSUE TITLE
[cxx-interop] support using libc++ on linux and windows

### ISSUE THREAD

**Issue by hyp** [apple/swift#69825]
Right now you can only use libstdc++ on linux and MSVC stl on windows, but we should support library like libc++ there too.

**Comment by compnerd** [comment]
Agreed, this would be nice (and something that I actually am in need of), but would require a set of flags.  I think that one other concern is that mixing and matching them would be bad.
